src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2031,6 +2060,28 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                 {
                     throw new ArgumentException(CodeAnalysisResources.StreamMustSupportWrite, nameof(pdbStream));
                 }
+
+                if (options?.EmitMetadataOnly == true)
+                {
+                    throw new ArgumentException(CodeAnalysisResources.PdbStreamUnexpectedWhenEmittingMetadataOnly, nameof(pdbStream));
+                }
+            }
+
+            if (metadataPeStream != null && options?.EmitMetadataOnly == true)
```

Comment:
(in that case pdbStream will be null since the PDB is emitted to peStream)

---
In reply to: [107311874](https://github.com/dotnet/roslyn/pull/17558#discussion_r107311874) [](ancestors = 107311874)